hoot-services/src/main/java/hoot/services/models/osm/Way.java

```diff
@@ -313,11 +313,36 @@ public void checkAndFailIfUsedByOtherObjects() throws OSMAPIAlreadyDeletedExcept
             List<Tuple> elementRecords = (List<Tuple>) Element.getElementRecordsWithUserInfo(getMapId(),
                     ElementType.Node, elementIds);
 
+            // It is too slow to do an XPathApi call for every single node to check if it
+            // already exists so its better to just get all nodes and store the ids in list
+            // and check against the list instead.
+            NodeList allNodesList;
+            try {
+                allNodesList = XPathAPI.selectNodeList(doc, "/osm/node");
+            }
+            catch (TransformerException e) {
+                throw new RuntimeException("Error invoking XPathAPI!", e);
+            }
+
+            ArrayList<String> allNodeIds = new ArrayList<>();
+            for(int i = 0; i< allNodesList.getLength(); i++) {
+                allNodeIds.add(allNodesList.item(i).getAttributes().getNamedItem("id").getNodeValue());
+            }
+
             for (Tuple elementRecord : elementRecords) {
-                Element nodeFullElement = ElementFactory.create(ElementType.Node, elementRecord, getMapId());
-                org.w3c.dom.Element nodeXml = nodeFullElement.toXml(parentXml, modifyingUserId,
-                        modifyingUserDisplayName, false, false);
-                parentXml.appendChild(nodeXml);
+                String nodeId = elementRecord.get(currentNodes).getId() + ""; // convert to string for cases of multiLayerUniqueElementIds
```

Comment:
the use of the query dsl `currentNodes` here gives me pause.  could be I just don't understand it's black magic, but we don't want to be making any db calls here.
we already have the node ids in both `nodeIds` and `elementIds` but I'm not sure their order will correspond with `elementRecords` so that we could use them.

Reply:
My only concern aside from the order is that above there is a call to `getElementRecordsWithUserInfo` which could cause a difference in the id lists